### ISSUE TITLE
Failed to activate protocol version: "CMAKE_HOME_DIRECTORY" is set but incompatible...

### ISSUE THREAD

**Issue by promethe42** [vector-of-bool/vscode-cmake-tools#508]
### Brief Issue Summary

```
[rollbar] Unhandled exception: Unhandled Promise rejection: configure [cmake-server] Failed to activate protocol version: "CMAKE_HOME_DIRECTORY" is set but incompatible with configured source directory value. {}
```

The CMake Project outline panel is empty.

### Expected:

No error.
The CMake Project outline panel is filled.

### Apparent Behavior:

```bash
git clone https://github.com/facebook/rocksdb.git
code rocksdb
```

CMake Tools will then attempt to "configure" the project and fail with the aforementioned error in the "debug console".

### Platform and Versions

- **Operating System**: Ubuntu 18.04 64bit
- **CMake Version**: 3.10.2
- **VSCode Version**: 1.25.1
- **CMake Tools Extension Version**: 1.1.1
- **Compiler/Toolchain**: gcc 7.3.0

**Comment by pointerless** [comment]
I had this same issue, I believe it is due to changing the project directory name/path. Try deleting the build directory and rerunning as loads of the stuff in that directory has absolute paths that are reused.

**Comment by promethe42** [comment]
As you can see in the issue description, it happens also with a freshly cloned repository.

**Comment by yay** [comment]
All my C++ projects created with the previous version of the plugin now have this issue. Is it possible for the plugin to recognize and fix whatever is wrong automatically?

**Comment by yay** [comment]
Removing the `build` directory and doing the `Clean reconfigure` works.

**Comment by promethe42** [comment]
It does work when the `build` directory is removed.
In my case, it was a permission issue: `build` is created - and build are done - from within a Docker container. So all the files are owned by `root`.

**Comment by vector-of-bool** [comment]
I've had this issue myself with Docker containers, as well as when you move the source/build directory. I know how it can be fixed, so I'll take a swing for 1.1.2.

**Comment by promethe42** [comment]
I'll try to create my container with the proper uid/gid to see if it fixes it.

**Comment by vector-of-bool** [comment]
This will now display a useful error message and helpful action button when it occurs. Fix in 1.1.2
